### DISCUSSION TITLE
Filtering target operations with regular expressions

### DISCUSSION THREAD

**Discussion by AliKhalili** [apideck-libraries/portman#251] [Q&A] [answered]
I want to add variation tests for those operations that have path parameters (`'/cars/{carId}'`). Could you please tell me what is the best way to do that?
I've supposed Portman accepts a valid regex for filtering the target operation. For example, I've referred to my targets operations with this regular expression:
`"openApiOperation": "*::/{.+}/",`
but it hasn't worked for me.
Thank you for your fantastic tool.

**Comment by thim81** [comment, marked as answer]
@AliKhalili We don't support regular expression, but we support wildcard.

We have it documented here: https://github.com/apideck-libraries/portman#portman-targeting

So your example could be achieved with: `openApiOperation": "*::/cars/*"`

**Reply by AliKhalili** [reply]
Thank you for your response. I should consider wildcard as an option, not regular expression for targeting operations.
So, in case I want to target all the operations with path parameters in a single variation test configuration, it seems I can not do that. Am I right?
I want to select all the operations with path parameters, like the following list of operations in one `openApiOperation` section:
`'/cars/{carId}'`,
`'/apps/{appName}/objects'`

**Reply by thim81** [reply]
@AliKhalili Not sure to fully understand the case but potentially you could define 2 Portman parts for each target:
```json
"contractTests": [
      {
        "openApiOperation": "*::/apps/*/objects",
        "statusSuccess": {
          "enabled": true
        },
       {
        "openApiOperation": "*::/cars/*",
        "statusSuccess": {
          "enabled": true
        }
      },
```

**Reply by AliKhalili** [reply]
Thanks

**Reply by thim81** [reply]
Your question triggered a feature request, to keep the configuration DRY.
We could introduce the feature to set an array of "openApiOperations" , like we have with "openApiOperationIds".

Idea: `"openApiOperations": ["*::/apps/*/objects","*::/cars/*"]`

**Reply by AliKhalili** [reply]
Cool, it would be a great idea and very handy. Is there any problem with supporting regular expression?
